src/services/db/__tests__/GitFileSystemService.spec.ts

```diff
@@ -57,6 +58,140 @@ describe("GitFileSystemService", () => {
     mockFs.restore()
   })
 
+  describe("listDirectoryContents", () => {
+    it("should return the contents of a directory successfully", async () => {
+      MockSimpleGit.cwd.mockReturnValueOnce({
+        revparse: jest.fn().mockResolvedValueOnce("another-fake-file-hash"),
+      })
+      MockSimpleGit.cwd.mockReturnValueOnce({
+        revparse: jest.fn().mockResolvedValueOnce("fake-dir-hash"),
+      })
+      MockSimpleGit.cwd.mockReturnValueOnce({
+        revparse: jest.fn().mockResolvedValueOnce("fake-empty-dir-hash"),
+      })
+
+      const expectedFakeDir: GitDirectoryItem = {
+        name: "fake-dir",
+        type: "dir",
+        sha: "fake-dir-hash",
+        path: "fake-dir",
+        size: 0,
+      }
+      const expectedFakeEmptyDir: GitDirectoryItem = {
+        name: "fake-empty-dir",
+        type: "dir",
+        sha: "fake-empty-dir-hash",
+        path: "fake-empty-dir",
+        size: 0,
+      }
+      const expectedAnotherFakeFile: GitDirectoryItem = {
+        name: "another-fake-file",
+        type: "file",
+        sha: "another-fake-file-hash",
+        path: "another-fake-file",
+        size: "Another fake content".length,
+      }
+
+      const result = await GitFileSystemService.listDirectoryContents(
+        "fake-repo",
+        ""
+      )
+      const actual = result
+        ._unsafeUnwrap()
+        .sort((a, b) => a.name.localeCompare(b.name))
```

Comment:
why is this sort important?

Reply:
This is the existing test behaviour - i moved the position of this test because it was failing for some reason haha

I think the sort is to ensure that the retrieval always returns in the same order, regardless of how it's returned

Reply:
Yup, to add on, jest also compares the order, even if the items are the same. This would ensure that the unit test does not get flaky for no reason.

Reply:
I see, I believe there is a jest function like strict equals or something to remove impact of order. But this looks good to me!